### PR TITLE
Update Docs - use cases

### PR DESCRIPTION
**Pull request by huanghao1998**
Add use cases: UC06, UC13, UC15
Edit use cases: UC03, UC04, UC12
Maintain consistency

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -370,7 +417,47 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-   Use case resumes at step 2.
+   Use case resumes from step 2.
+
+**Use case: UC13 Undo recent delete/clear operation**
+
+**MSS**
+
+1. User chooses to delete/clear contact(s).
+2. SOCket deletes respective contact(s).
+3. User regrets and requests to get back the recent deleted contact(s).
```
Comment (dillongoh) [inline docs/DeveloperGuide.md:428]: I don't think need to mention the user "regrets"

**Review by dillongoh — COMMENTED** [review]
Other than the word "regret" seeming a bit out of place the rest is good to merge.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ title: Developer Guide
 
 ## **Acknowledgements**
 
-* This project is based on the AddressBook-Level3 project created by the [SE-EDU initiative](https://se-education.org).
+* {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
```
Comment (dillongoh) [inline docs/DeveloperGuide.md:12]: Please revert it back to the addressbook3 reference. That’s supposed to be there.

Comment (huanghao1998) [reply docs/DeveloperGuide.md]: noted with thanks!

**Pull request merged**